### PR TITLE
576 Enable email notifications and configuration for study enquiries

### PR DESCRIPTION
**Pull request by kevinhowbrook**
Resolves ticket [576](https://projects.torchbox.com/projects/rca-website-rebuild/tickets/576)

image

### Testing:

- Go to https://rca-development.herokuapp.com/admin/settings/enquire_to_study/enquiretostudysettings/2/ to configure the email settings and save them
- Got to https://rca-development.herokuapp.com/enquire-to-study/ and fill out the form using a real email address.
- Check your email inbox and spam folders to confirm email 

This is using the EMAIL_HOST_USER as the `from` address. On the development servier this will be `generic@staging.torchbox.com`. On RCA production the from address will be `postmaster@rca.ac.uk` so that may need changing.

After submitting the form you should receive an email to the address filled out in the forms 'email' field. Please be sure to check your spam.

### CODE REVIEW

File: rca/enquire_to_study/views.py [hunk 1]
```diff
@@ -33,6 +36,32 @@ def post_qs(self, form_data):
     def create_form_submission(self, form):
         form.save()
 
+    def send_user_email_notification(self, form):
+        enquiry_form_settings = EnquireToStudySettings.objects.first()
+        if not enquiry_form_settings.email_submission_notifations:
```
Comment (kevinhowbrook) [inline rca/enquire_to_study/views.py:41]: AC: Automatic responses for Enquiry forms can be toggle on or off

File: rca/enquire_to_study/models.py [hunk 2]
```diff
@@ -154,3 +156,37 @@ class EnquiryFormSubmissionProgrammesOrderable(Orderable):
     panels = [
         SnippetChooserPanel("programme"),
     ]
+
+
+@register_setting
```
Comment (kevinhowbrook) [inline rca/enquire_to_study/models.py:161]: AC: As an administrator I can define copy and subject line for the email notification

File: rca/enquire_to_study/views.py [hunk 3]
```diff
@@ -33,6 +36,32 @@ def post_qs(self, form_data):
...
+        # add links to programmes for the response
+        if "programmes" in form.cleaned_data:
+            email_content += (
+                "\n\nPlease see the following pages for more "
+                "information about the courses you enquired about: \n"
+            )
+            for programme in form.cleaned_data["programmes"]:
+                email_content += (
+                    f"{programme.title} {settings.BASE_URL + programme.url} \n"
+                )
```
Comment (kevinhowbrook) [inline rca/enquire_to_study/views.py:53]: AC: As a user, if I have enquired about a programme, then my email notification includes a link to the relevant programme page(s) for the programme(s) i have selected

File: rca/enquire_to_study/views.py [hunk 4]
```diff
@@ -33,6 +36,32 @@ def post_qs(self, form_data):
     def create_form_submission(self, form):
         form.save()
 
+    def send_user_email_notification(self, form):
+        enquiry_form_settings = EnquireToStudySettings.objects.first()
```
Comment (richbrennan) [inline rca/enquire_to_study/views.py:40]: Is this site single or multi-site? `EnquireToStudySettings.objects.first()` will only return the correct settings for one site in a multi-site setup, in which case you'd use https://docs.wagtail.io/en/stable/reference/contrib/settings.html#using-in-python

Comment (kevinhowbrook) [reply rca/enquire_to_study/views.py]: Thanks, yea this was lazy on my part. It's one site but in the future there may be more. Fixed in https://github.com/torchbox/rca-wagtail-2019/pull/586/commits/68ca60e026ee71f25662f224b53e24e6ca8d593d

**Pull request merged**